### ISSUE TITLE
run_fg() got an unexpected keyword argument

### ISSUE THREAD

**Issue by xapple** [tomerfiliba/plumbum#441]
I'm trying to redirect the `stdout` and `stderr` of a subprocess to different log files. Though the documentation doesn't mention this directly, I was able to figure out the following syntax:

     import plumbum
     cmd = plumbum.local['super_program.py']
     ((cmd > 'stdout.txt') >= 'stderr.txt')()

Of course this syntax, trying to emulate standard shell grammar, is highly unpythonic. I thought I would get away with this:

     cmd.run_fg(stdout='stdout.txt', stderr='stderr.txt')

But this raises a TypeError? Is that expected behavior? How can I specify the IO redirection without awkward operator overloading? In the `sh` module for instance you would just do `cmd(_out="stdout.txt", _err="stderr.txt")`

**Comment by AndydeCleyre** [comment]
I can't reach a computer right now, but you may need to pass a file-like or stream-like object, rather than a string. Try passing the opened objects instead of the file names.

**Comment by AndydeCleyre** [comment]
Additionally, since you're redirecting the out and err, it's not a foreground operation, but a background one. So that's why the kwargs aren't available for `run_fg`.

```python
from plumbum import local
local['ls'].run_bg(stdout=local.path('stdout.txt').open('a'), stderr=local.path('stderr.txt').open('a'))
```

**Comment by xapple** [comment]
Thanks for your response.

I thought the difference between background and foreground only concerned synchronicity. If I want to block until the completion of the call is not hence a foreground process independently of any redirection occurring?

That example shows the syntax I could use in detail, thanks. But I think it's a pity I have to provide the file objects myself. The other calls such as ` ((cmd > 'stdout.txt') >= 'stderr.txt')()` are able to use file paths directly. Is this not the point of such a helper library to reduce the boilerplate? I'm still debating about staying with `sh` (but that other library doesn't work under Windows+Python3).

**Comment by AndydeCleyre** [comment]
> If I want to block until the completion of the call is it not hence a foreground process independently of any redirection occurring?

No, I don't think so. For example, calling a cmd with parentheses blocks, but is not run in the foreground (in the bash sense). That said, I don't see any harm in adding the redirection kwargs to `run_fg`, but I'm not a maintainer here.

I also think it would be reasonable to push all `str` redirection kwargs though `local.path(...).open('a')`, assuming that's what happens using the equivalent symbolic syntax.

**Comment by xapple** [comment]
Thanks a lot for the explanation, yes I agree with all your points.
